src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -0,0 +1,571 @@
+        env, vol = self.setup_callback_server_envs(job_dir, kb_base_url, token, port)
+        self.container = client.containers.run(
+            name=container_name,
+            image=loader_common_names.CALLBACK_UPLOADER_IMAGE_NAME,
+            detach=True,
+            network_mode="host",
+            environment=env,
+            volumes=vol,
+        )
+        time.sleep(2)
+
+    def stop_callback_server(self):
+        self.container.stop()
+        self.container.remove()
+
+
+def _get_parser():
+    parser = argparse.ArgumentParser(
+        description="PROTOTYPE - Upload assembly files to the workspace service (WSS).",
+        formatter_class=loader_helper.ExplicitDefaultsHelpFormatter,
+    )
+
+    required = parser.add_argument_group("required named arguments")
+    optional = parser.add_argument_group("optional arguments")
+
+    # Required flag argument
+    required.add_argument(
+        "--workspace_id",
+        required=True,
+        type=int,
+        help="Target workspace addressed by the permanent ID",
+    )
+    required.add_argument(
+        f"--{loader_common_names.KBASE_COLLECTION_ARG_NAME}",
+        type=str,
+        help="The name of the collection being processed",
+    )
+    required.add_argument(
+        f"--{loader_common_names.SOURCE_VER_ARG_NAME}",
+        type=str,
+        help="The source version of the collection being processed",
+    )
+
+    # Optional argument
+    optional.add_argument(
+        "--root_dir",
+        type=str,
+        default=loader_common_names.ROOT_DIR,
+        help="Root directory.",
+    )
+    optional.add_argument(
+        "--token_filepath",
+        type=str,
+        help="A file path that stores a KBase token",
+    )
+    optional.add_argument(
+        "--env",
+        type=str,
+        choices=loader_common_names.KB_ENV,
+        default="PROD",
+        help="KBase environment, defaulting to PROD",
+    )
+    optional.add_argument(
+        "--upload_file_ext",
+        type=str,
+        nargs="+",
+        default=UPLOAD_FILE_EXT,
+        help="Upload only files that match given extensions",
+    )
+    optional.add_argument(
+        "--workers",
+        type=int,
+        default=5,
+        help="Number of workers for multiprocessing",
+    )
+    optional.add_argument(
+        "--keep_job_dir",
+        action="store_true",
+        help="Keep SDK job directory after upload task is completed",
+    )
+    return parser
+
+
+def _get_yaml_file_path(assembly_dir: str) -> str:
+    """
+    Get the uploaded.yaml file path from collections source directory.
+    """
+    file_path = os.path.join(assembly_dir, UPLOADED_YAML)
+    Path(file_path).touch(exist_ok=True)
+    return file_path
+
+
+def _sanitize_data_dir(job_dir: str) -> str:
+    """
+    Create a temporary directory for storing uploaded files.
+    """
+    data_dir = os.path.join(job_dir, "workdir/tmp", DATA_DIR)
+    if os.path.exists(data_dir):
+        shutil.rmtree(data_dir)
```

Comment:
If 2 instances of the uploader were running at once this could get ugly. Maybe ok for now but might need a temp directory that's unique per instance

Reply:
added uuid str as part of dir name.

Reply:
What if, say, myself and Tian are both running the uploader at the same time? Then one of our token files will overwrite the other

Reply:
will not as you and Tian have different usernames.

collections/sdk_job_dir/`<username>`/workdir/tmp/`data_dir_<uuid>`

Reply:
Right, should've known that. Ok, what if you're uploading to CI and prod at the same time? One of the tokens will overwrite the other

Reply:
Let's say I upload to CI and prod at the same time, uuid4 are still different. So all files are still save in different temp dir. 
But it sounds like you wanna add `env` to the path as well to be safe.

Reply:
> So all files are still saved in different temp dirs.

Not the ones that are saved in workdir or workdir/tmp. For example:

```
crushingismybusiness@andbusinessisgood:~/github/kbaseapps/AssemblyUtil/test_local/workdir$ ls
config.properties  coverage.xml  test.cfg  test_coverage  token
crushingismybusiness@andbusinessisgood:~/github/kbaseapps/AssemblyUtil/test_local/workdir$ cat config.properties 
[global]
kbase_endpoint = https://ci.kbase.us/services
job_service_url = https://ci.kbase.us/services/userandjobstate
workspace_url = https://ci.kbase.us/services/ws
shock_url = https://ci.kbase.us/services/shock-api
handle_url = https://ci.kbase.us/services/handle_service
srv_wiz_url = https://ci.kbase.us/services/service_wizard
njsw_url = https://ci.kbase.us/services/njs_wrapper
auth_service_url = https://ci.kbase.us/services/auth/api/legacy/KBase/Sessions/Login
auth_service_url_allow_insecure = false
```

Reply:
> But it sounds like you wanna add env to the path as well to be safe?

I would just move the UUID from the end of the path to part of the job_dir path, so maybe <skd_job_dir>/<user_id>/<uuid>

Reply:
Updated!

Reply:
:+1: